gitlab-runner-mock/src/api/artifacts.rs

```diff
@@ -44,47 +45,52 @@ impl Respond for JobArtifactsUploader {
                     .get(&CONTENT_TYPE)
                     .expect("Missing content type");
 
-                let boundary = ct
-                    .to_str()
-                    .expect("Invalid content type header value")
-                    .split_once("boundary=")
-                    .map(|x| x.1)
-                    .expect("Missing boundary");
-
-                let mut multipart =
-                    Multipart::with_body(std::io::Cursor::new(&request.body), boundary);
+                let boundary = ct.to_str().expect("Invalid content type header value");
+                let boundary = multer::parse_boundary(boundary).expect("Failed to parse boundary");
+                let mut multipart = Multipart::new(
+                    futures::stream::iter(
+                        request
+                            .body
+                            .chunks(10254)
```

Comment:
Where is this number from? It doesn't seem to be a round number of KiB or anything like that?